Composed/Core/DataSources/ComposedDataSource.swift

```diff
@@ -284,80 +270,25 @@ private extension ComposedDataSource {
 
 extension ComposedDataSource: DataSourceUpdateDelegate {
 
-    public final func dataSource(_ dataSource: DataSource, didInsertSections sections: IndexSet) {
+    public func dataSource(_ dataSource: DataSource, performUpdates changeDetails: ComposedChangeDetails) {
+        var details = ComposedChangeDetails(hasIncrementalChanges: changeDetails.hasIncrementalChanges)
         let mapping = self.mapping(for: dataSource)
-        let global = sections.map(mapping.globalSection(forLocal:))
-        _invalidate()
-        updateDelegate?.dataSource(self, didInsertSections: IndexSet(global))
-    }
 
-    public final func dataSource(_ dataSource: DataSource, didDeleteSections sections: IndexSet) {
-        let mapping = self.mapping(for: dataSource)
-        let global = sections.map(mapping.globalSection(forLocal:))
-        _invalidate()
-        updateDelegate?.dataSource(self, didDeleteSections: IndexSet(global))
-    }
-
-    public final func dataSource(_ dataSource: DataSource, didUpdateSections sections: IndexSet) {
-        let mapping = self.mapping(for: dataSource)
-        let global = sections.map(mapping.globalSection(forLocal:))
-        updateDelegate?.dataSource(self, didUpdateSections: IndexSet(global))
-        _invalidate()
-    }
+        details.insertedSections = IndexSet(changeDetails.insertedSections.map(mapping.globalSection(forLocal:)))
+        details.insertedIndexPaths = changeDetails.insertedIndexPaths
+        details.removedSections = IndexSet(changeDetails.removedSections.map(mapping.globalSection(forLocal:)))
+        details.removedIndexPaths = changeDetails.removedIndexPaths
+        details.updatedSections = IndexSet(changeDetails.updatedSections.map(mapping.globalSection(forLocal:)))
+        details.updatedIndexPaths = changeDetails.updatedIndexPaths
+        details.movedSections = changeDetails.movedSections.map(mapping.globalSections(forLocal:))
+        details.movedIndexPaths = changeDetails.movedIndexPaths.map(mapping.globalIndexPaths(forLocal:))
```

Comment:
It looks like these could be moved in to an initialiser for the `ComposedChangeDetails`?